metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala

```diff
@@ -158,21 +158,28 @@ final class FormattingProvider(
     }
   }
 
+  private var dontAskFileCreation: Boolean = false
```

Comment:
This var won't persist if the Metals server restarts. Normally we implement "Don't show again" by persisting the user's request in a sql database. Example: https://github.com/scalameta/metals/blob/e2320b5b4d7f365c6064ccc3ecb2a105b579a109/metals/src/main/scala/scala/meta/internal/metals/DismissedNotifications.scala

Reply:
That's intentional behavior. If we persist it, we have to provide a way to re-enable it too. So, I thought asking once in each session should be better. I didn't know that we can configure formatting behavior per workspace.

Reply:
It can be unexpected for the user to select "Don't show again", restart VS Code and then get the .scalafmt.conf prompt again.

I agree it would be nice to have a way to re-enable prompts where the user selected "Don't show again", the only way to achieve that right now is to `rm -rf .metals`. This functionality be implemented separately from this PR and it would then work for all "Don't show again" prompts.

Reply:
How about asking "Not now" instead?

Reply:
We also use the SQL database to handle "Not now" responses to dismiss this prompt for a X number of minutes. For example, here we handle "Not now" for the "Import build" prompt https://github.com/scalameta/metals/blob/abca8703638fe2768e0058288ae0e3decbe42e07/metals/src/main/scala/scala/meta/internal/metals/BloopInstall.scala#L163

> Being asked every time when I save files even after selecting "Not now" is annoying. I just want to avoid it.

I agree this is annoying and I'm open to merge this feature but I'm not convinced it's a reliable fix to store this state in the Metals server process because that means the user gets the prompt again when they restart Metals. The reliable fix for this problem is to use the SQL table, which is consistent with how we handle other prompts